packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+const List<String> unsupportedVersions = <String>[
+  '6',
+  '7',
+  '8',
+];
+
+/// Validator to be run with `flutter doctor` to check
+/// Windows host machines if they are running supported versions.
+class WindowsVersionValidator extends DoctorValidator {
+  const WindowsVersionValidator({required ProcessManager processManager})
+      : _processManager = processManager,
+        super('Windows Version');
+
+  final ProcessManager _processManager;
+
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult result;
+    try {
+      result = await _processManager.run(<String>['systeminfo']);
+    } on ProcessException {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[],
+        statusInfo:
+            'Unable to run Windows version check using built-in `systeminfo`',
+      );
+    }
+
+    if (result.exitCode != 0) {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[],
+        statusInfo: 'Exit status from running `systeminfo` was unsuccessful',
+      );
+    }
+
+    final String resultStdout = result.stdout as String;
+
+    // Regular expression pattern for identifying
+    // semantic versioned strings
+    // (ie. 10.5.4123)
+    final RegExp regex =
```

Comment:
I would pull this into a @visibleForTesting method that takes the stdout string and then returns `Iterable<RegExpMatch> matches`, and then you can unit test different strings